src/theory/fp/fp_converter.cpp

```diff
@@ -820,868 +820,469 @@ FpConverter::uf FpConverter::buildComponents(TNode current)
   return tmp;
 }
 
-// Non-convertible things should only be added to the stack at the very start,
-// thus...
-#define CVC5_FPCONV_PASSTHROUGH Assert(workStack.empty())
-
 Node FpConverter::convert(TNode node)
 {
-  std::vector<TNode> workStack;
-  TNode result = node;
-
-  workStack.push_back(node);
+  std::vector<TNode> visit;
+  std::unordered_map<TNode, bool> visited;
+  NodeManager* nm = NodeManager::currentNM();
 
-  NodeManager *nm = NodeManager::currentNM();
+  visit.push_back(node);
 
-  while (!workStack.empty())
+  while (!visit.empty())
   {
-    TNode current = workStack.back();
-    workStack.pop_back();
-    result = current;
+    TNode cur = visit.back();
+    visit.pop_back();
+    TypeNode t(cur.getType());
+
+    if ((t.isBoolean() && d_boolMap.find(cur) != d_boolMap.end())
+        || (t.isRoundingMode() && d_rmMap.find(cur) != d_rmMap.end())
+        || (t.isBitVector() && d_sbvMap.find(cur) != d_sbvMap.end())
+        || (t.isBitVector() && d_ubvMap.find(cur) != d_ubvMap.end())
+        || (t.isFloatingPoint() && d_fpMap.find(cur) != d_fpMap.end()))
+    {
+      continue;
+    }
```

Comment:
If this condition is true, then we have already word-blasted `cur`, right? Maybe add a comment.